frontend/src/app/services/auth/auth.guard.ts

```diff
@@ -27,7 +27,7 @@ export class AuthGuard implements CanActivate, CanActivateChild {
     if (currentUser && currentUser.token) {
       if (this.helper.isTokenExpired(currentUser.token)) {
         this.authService.refreshToken().pipe(first()).subscribe(isValid => {
-          return isValid;
+          return isValid !== null;
```

Comment:
The reason we need to double-click on things when our refresh token expires is because this observable has no subscribers. This `canActivate()` should probably return an `Observable<boolean>`. https://stackoverflow.com/a/50143721/1639835 seems to have a good idea here.